### ISSUE TITLE
Adding a new Global Secondary Index to an existing DynamoDB Table Prompts Terraform to recreate existing GSIs.

### ISSUE THREAD

**Issue by GL-KenDLG** [terraform-providers/terraform-provider-aws#12928]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.24
+ provider.aws v2.54.0
+ provider.external v1.2.0
+ provider.kubernetes v1.11.1
+ provider.null v2.1.2
+ provider.template v2.1.2

### Affected Resource(s)

aws_dynamodb_table

### Terraform Configuration Files

Posting relevant section of the config.

```  
global_secondary_index {
    name            = "external_reference-external_timestamp-index"
    hash_key        = "external_reference"
    range_key       = "external_timestamp"
    projection_type = "ALL"
    read_capacity   = 0
    write_capacity  = 0
  }

  global_secondary_index {
    name            = "journal_id-index"
    hash_key        = "journal_id"
    projection_type = "ALL"
    read_capacity   = 0
    write_capacity  = 0
  }

  global_secondary_index {
    name            = "journal_reference-index"
    hash_key        = "journal_reference"
    projection_type = "ALL"
    read_capacity   = 0
    write_capacity  = 0
  }

  global_secondary_index {
    name            = "original_external_reference-external_timestamp-index"
    hash_key        = "original_external_reference"
    range_key       = "external_timestamp"
    projection_type = "ALL"
    read_capacity   = 0
    write_capacity  = 0
  }

  global_secondary_index {
    name            = "hourly-index"
    hash_key        = "hourly"
    projection_type = "INCLUDE"
    non_key_attributes = ["journal_reference","created_at"]
    read_capacity   = 0
    write_capacity  = 0
  }
```
Note: The new index is the one at the bottom. I have tried placing it in different places in the declaration but it still did not resolve this issue.

### Expected Behavior

terraform apply should add the new global secondary index and leave the others existing indexes alone.

### Actual Behavior
This is an existing table with existing GSIs and data. If I attempt to add a new GSI, TF wants to recreate 3 out of the 5 existing GSIs.

This is a partial output of a `terraform plan` where you can see it wants to recreate most of the existing GSIs

```
        global_secondary_index {
            hash_key           = "external_reference"
            name               = "external_reference-external_timestamp-index"
            non_key_attributes = []
            projection_type    = "ALL"
            range_key          = "external_timestamp"
            read_capacity      = 0
            write_capacity     = 0
        }
      + global_secondary_index {
          + hash_key           = "hourly"
          + name               = "hourly-index"
          + non_key_attributes = [
              + "journal_reference",
              + "created_at",
            ]
          + projection_type    = "INCLUDE"
          + read_capacity      = 0
          + write_capacity     = 0
        }
      - global_secondary_index {
          - hash_key           = "journal_id" -> null
          - name               = "journal_id-index" -> null
          - non_key_attributes = [] -> null
          - projection_type    = "ALL" -> null
          - read_capacity      = 0 -> null
          - write_capacity     = 0 -> null
        }
      + global_secondary_index {
          + hash_key           = "journal_id"
          + name               = "journal_id-index"
          + non_key_attributes = []
          + projection_type    = "ALL"
          + read_capacity      = 0
          + write_capacity     = 0
        }
      - global_secondary_index {
          - hash_key           = "journal_original_reference" -> null
          - name               = "journal_original_reference-index" -> null
          - non_key_attributes = [] -> null
          - projection_type    = "ALL" -> null
          - read_capacity      = 0 -> null
          - write_capacity     = 0 -> null
        }
      + global_secondary_index {
          + hash_key           = "journal_original_reference"
          + name               = "journal_original_reference-index"
          + non_key_attributes = []
          + projection_type    = "ALL"
          + read_capacity      = 0
          + write_capacity     = 0
        }
      - global_secondary_index {
          - hash_key           = "journal_reference" -> null
          - name               = "journal_reference-index" -> null
          - non_key_attributes = [] -> null
          - projection_type    = "ALL" -> null
          - read_capacity      = 0 -> null
          - write_capacity     = 0 -> null
        }
      + global_secondary_index {
          + hash_key           = "journal_reference"
          + name               = "journal_reference-index"
          + non_key_attributes = []
          + projection_type    = "ALL"
          + read_capacity      = 0
          + write_capacity     = 0
        }
        global_secondary_index {
            hash_key           = "original_external_reference"
            name               = "original_external_reference-external_timestamp-index"
            non_key_attributes = []
            projection_type    = "ALL"
            range_key          = "external_timestamp"
            read_capacity      = 0
            write_capacity     = 0
        }
```
### Steps to Reproduce

1. Create a aws_dynamo_table with a few GSI
2. Attempt to add a new GSI

### References

* #3828 <- adding read_capacity and write_capacity fixed one of the GSIs but not the others.

**Comment by milkydood** [comment]
I've just run into this very same issue, where terraform plan says it's going to re-create a whole bunch of indexes, when it should just be adding one.

As far as I can tell though, it actually does just add the single index on an apply, despite what the plan says.

**Comment by elblivion** [comment]
Also seeing this, also when removing a GSI. Terraform v0.12.24 + provider.aws v3.0.0. Applying such a change led to all GSIs being destroyed and re-creation failing. :-/

**Comment by maryelizbeth** [comment]
Hi Y'all, we are currently tracking this issue via #671.

We've added #671 to our team's prioritization board and are planning to address it and its related PRs in a near-term release. As this issue is a duplicate, we'll close this and ask that y'all continue discussion on #671. If you believe this has been closed in error or that #671 does not adequately address the issue, please open a new issue.

Thanks!
